### PR TITLE
FourSumProblem #4374

### PR DESCRIPTION
**Pull request by Shonu72**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/main/java/com/thealgorithms/misc/FourSumProblem.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.thealgorithms.misc;
+import java.util.*;
+
+public class FourSumProblem {
+    public static List<List<Integer>> fourSum(int[] nums, int target) {
+        int n = nums.length; // size of the array
+        Set<List<Integer>> set = new HashSet<>();
+
+        // checking all possible quadruplets:
+        for (int i = 0; i < n; i++) {
```
Comment (BamaCharanChhandogi) [inline src/main/java/com/thealgorithms/misc/FourSumProblem.java:10]: This approach does not have a good time complexity. Try to achieve better time complexity

**Review by BamaCharanChhandogi — CHANGES_REQUESTED** [review]
Add all possible test cases.👍

**Comment by vil02** [comment]
Is it a duplicate of #4375 and #4377?

**Comment by BamaCharanChhandogi** [comment]
I am closing this PR.

**Pull request closed without merging**